### PR TITLE
Using -o, new valestrom binary, build changes

### PR DESCRIPTION
**Pull request by Verdagon**
Updated valestrom binary, using the most recent version now.
Made tests use -o option, simplifying them a bit.
Cut down components list, and changed code to not depend on / initialize so many.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,144 +29,34 @@ include_directories(
 )
 add_definitions(${LLVM_DEFINITIONS})
 
-# output from `llvm-config --components | tr " " "\n"`
+include_directories("${CMAKE_SOURCE_DIR}/src/c-compiler/")
+
+link_directories(/usr/lib/llvm-7/lib)
```
Comment (Verdagon) [inline CMakeLists.txt]: Try removing, or pulling out into an env variable or something

File: test/test_vale.py [hunk 2]
```diff
@@ -16,60 +16,29 @@ def procrun(args: List[str], **kwargs) -> subprocess.CompletedProcess:
...
-        vir = json.loads(vir_str)
-        cls.write_json_to_vir_file(vir, fp=output_file)
 
     def valec(self, args: List[str]) -> subprocess.CompletedProcess:
         assert self.GENPATH
         return procrun([f"../{self.GENPATH}/valec"] + args)
 
     def clang(self, args: List[str]) -> subprocess.CompletedProcess:
-        return procrun(["clang"] + args)
+        return procrun(["/usr/bin/clang-7"] + args)
```
Comment (Verdagon) [inline test/test_vale.py]: Remove this path somehow

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,13 +2,12 @@ cmake_minimum_required(VERSION 3.11)
 project(valec)
 cmake_policy(VERSION 3.11)
 
-find_package(LLVM 7.1.0 REQUIRED CONFIG)
+find_package(LLVM 7 REQUIRED CONFIG)
```
Comment (Verdagon) [inline CMakeLists.txt:5]: Somehow pull this out into a var, or require either 7 or 7.1

Comment (Verdagon) [inline CMakeLists.txt:5]: Looks like this should work with 7.1.0 because we dont specify EXACT

**Review by solstice333 — APPROVED** [review]

**Pull request merged**